src/univ/listbox.cpp

```diff
@@ -1085,27 +1078,22 @@ void wxListBox::DoSelect(int item, bool sel)
 
 void wxListBox::SelectAndNotify(int item)
 {
-    DoSelect(item);
-
-    SendEvent(wxEVT_LISTBOX);
+    if( item != -1 )
+    {
+        DoSelect(item);
+        SendEvent(wxEVT_LISTBOX);
+    }
 }
 
 void wxListBox::Activate(int item)
 {
     if ( item != -1 )
-        SetCurrentItem(item);
-    else
-        item = m_current;
```

Comment:
If this is not done here, it should probably be done in `PerformAction()`, as I see that it does it for other actions. I must admit I don't remember at all why is it so, but it looks like it's valid to call it with invalid item.

Reply:
I agree that click on non-item probably should just deselect current item without selecting any item (as far I understand that was expected initial behaviour and it's kind of logical) but at least MSW-port just do nothing (it's also logical) and that is why I changed behaviour here to match it. 

I think that current mess in code is due to comment that said that -1 is passed as current item instead of invalid item. And as earlier HitTest always returned "correct" item index, any click on non-item area desecelected current and re-selected it again that obvious doesn't make any sense.